provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1798,6 +1812,108 @@ func generateSchema() schema.PackageSpec {
 					Required: []string{"content", "contentType"},
 				},
 			},
+			"eks:index:ResolveConflictsOnUpdate": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "string",
+					Description: "How to resolve field value conflicts for an Amazon EKS add-on if you've changed a value from the Amazon EKS default value. " +
+						"Valid values are `NONE`, `OVERWRITE`, and `PRESERVE`. For more details see the [UpdateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_UpdateAddon.html) API Docs.",
+				},
+				Enum: []schema.EnumValueSpec{
+					{
+						Name:        "None",
+						Value:       "NONE",
+						Description: "Amazon EKS doesn't change the value. The update might fail.",
+					},
+					{
+						Name:        "Overwrite",
+						Value:       "OVERWRITE",
+						Description: "Amazon EKS overwrites the changed value back to the Amazon EKS default value.",
+					},
+					{
+						Name:        "Preserve",
+						Value:       "PRESERVE",
+						Description: "Amazon EKS preserves the value. If you choose this option, we recommend that you test any field and value changes on a non-production cluster before updating the add-on on your production cluster.",
+					},
+				},
+			},
+			"eks:index:ResolveConflictsOnCreate": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "string",
+					Description: "How to resolve field value conflicts when migrating a self-managed add-on to an Amazon EKS add-on. " +
+						"Valid values are `NONE` and `OVERWRITE`. For more details see the [CreateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateAddon.html) API Docs.",
+				},
+				Enum: []schema.EnumValueSpec{
+					{
+						Name:        "None",
+						Value:       "NONE",
+						Description: "If the self-managed version of the add-on is installed on your cluster, Amazon EKS doesn't change the value. Creation of the add-on might fail.",
+					},
+					{
+						Name:  "Overwrite",
+						Value: "OVERWRITE",
+						Description: "If the self-managed version of the add-on is installed on your cluster and the Amazon EKS default value is different than the existing value, " +
+							"Amazon EKS changes the value to the Amazon EKS default value.",
+					},
+				},
+			},
+			"eks:index:CoreDnsAddonOptions": {
```

Comment:
We should add the `serviceAccountRoleArn` prop as well.

I'm also wondering if we should allow users to pass configuration values and then merge those with the ones the provider sets.

Modifying the resource requests/limits or replica count of coredns is a common config for example.

Reply:
But we can exclude that from this PR here. I've added it to the VPC CNI and if we both start adding those settings we're just gonna step on each others toes

Reply:
> We should add the serviceAccountRoleArn prop as well.

I don't think either `coredns` or `kube-proxy` use a role. At least according to [this doc](https://docs.aws.amazon.com/eks/latest/userguide/workloads-add-ons-available-eks.html#add-ons-coredns)

Reply:
Ah nice! Didn't know that